### DISCUSSION TITLE
second order conic programming with quadratic objective function

### DISCUSSION THREAD

**Discussion by mamammiya** [coin-or/Ipopt#462] [Q&A] [answered]
I am trying to solve a convex optimization problem using the IPOPT:

minx  x'Hx + f'x  + (x'Rx)^1/2
s.t.      Ax<=b

or equivalently,

 minx  x'Hx + f'x  + norm(chol(R)*x)
s.t.      Ax<=b

where H is a positive semi-definite matrix, f<0 and R is a positive semi-definite diagonal matrix. I was wondering, is there any way I could transfer my problem into a more "IPOPT friendly" format?  Otherwise, I think I have to directly provide the Jocobian and Hessian matrices of norm(.)

Thanks!!

**Comment by svigerske** [comment, marked as answer]
Ipopt itself doesn't come with any features to support modeling of optimization problems.

Ipopt has an interface to read .nl files included, so it can be used from AMPL, but also other software that communicates via .nl files, e.g., Pyomo. Further, https://coin-or.github.io/Ipopt/index.html#HOWTOUSE lists a number of software packages that have interfaces to Ipopt.
